src/electrum.rs

```diff
@@ -401,6 +401,35 @@ impl Rpc {
         }
     }
 
+    fn transaction_from_pos(
+        &self,
+        (height, tx_pos, merkle): &(usize, usize, bool),
+    ) -> Result<Value> {
+        let chain = self.tracker.chain();
+        let blockhash = match chain.get_block_hash(*height) {
+            None => bail!("missing block at {}", height),
+            Some(blockhash) => blockhash,
+        };
+        let txids = self.daemon.get_block_txids(blockhash)?;
+        let txid: Txid = txids[*tx_pos];
+        if *merkle {
+            match txids.iter().position(|current_txid| *current_txid == txid) {
+                None => bail!("missing txid {} in block {}", txid, blockhash),
```

Comment:
Not sure that we may get here, since `txid == txids[*tx_pos]`, so `position == *tx_pos`...

Reply:
Added test for invalid `tx_pos` in the latest commit.

Reply:
Pushed a small simplification commit (https://github.com/romanz/electrs/pull/836/commits/5928b990442ee50f6df71af0db38b44fdb08ddd4) - please take a look :)